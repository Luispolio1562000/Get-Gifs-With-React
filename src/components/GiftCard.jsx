import PropTypes from "prop-types";

function GiftCard({ title, url }) {
  return (
    <div className="card">
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  );
}

GiftCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
export default GiftCard;
