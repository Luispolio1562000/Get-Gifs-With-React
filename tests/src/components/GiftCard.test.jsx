import { render, screen } from "@testing-library/react";
import GiftCard from "../../../src/components/GiftCard";

describe("Tests <GiftCard />", () => {
  const title = "Gift Card";
  const url = "https://github.com/";
  test("Should match to Snapshot", () => {
    const { container } = render(<GiftCard title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("should show the image with the URL and ALT indicated", () => {
    render(<GiftCard title={title} url={url} />);
    //screen.debug();
    //* Html IMG element.
    // console.log(screen.getByRole("img").src);
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show the title in the component", () => {
    render(<GiftCard title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
