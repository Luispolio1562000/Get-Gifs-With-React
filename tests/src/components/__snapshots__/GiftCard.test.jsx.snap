// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests <GiftCard /> Should match to Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
       
      Gift Card
    </p>
    <img
      alt="Gift Card"
      src="https://github.com/"
    />
  </div>
</div>
`;
